File: codewars/6kyu/find-the-missing-letter/dduck-find-the-missing-letter.js
// find-the-missing-letter
// https://www.codewars.com/kata/5839edaa6754d6fec10000a2/train/javascript

const findMissingLetter = (array) => array.reduce((acc, cur, i, arr) => {
  if (acc.charCodeAt() + 1 === cur.charCodeAt()) {
    return cur;
  } else {
    const result = acc.charCodeAt() + 1;
    arr.splice(0);
    return String.fromCharCode(result);
  }
}, String.fromCharCode(array[0].charCodeAt() - 1));

console.log(findMissingLetter(['a', 'b', 'c', 'd', 'f']), 'e');
console.log(findMissingLetter(['O', 'Q', 'R', 'S']), 'P');

File: codewars/6kyu/find-the-missing-letter/dduck-find-the-missing-letter.py
# find-the-missing-letter
# https://www.codewars.com/kata/5839edaa6754d6fec10000a2/train/javascript

from unittest import TestCase


def find_missing_letter(chars):
    prev = chars[0]

    for i in range(1, len(chars)):
        if ord(prev) + 1 != ord(chars[i]):
            return chr(ord(prev) + 1)
        prev = chars[i]


TestCase().assertEqual(find_missing_letter(['a', 'b', 'c', 'd', 'f']), 'e')
TestCase().assertEqual(find_missing_letter(['O', 'Q', 'R', 'S']), 'P')
